hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieBootstrapConfig.java

```diff
@@ -67,13 +67,6 @@ public class HoodieBootstrapConfig extends HoodieConfig {
       .sinceVersion("0.6.0")
       .withDocumentation("Selects the mode in which each file/partition in the bootstrapped dataset gets bootstrapped");
 
-  public static final ConfigProperty<String> DATA_QUERIES_ONLY = ConfigProperty
-      .key("hoodie.bootstrap.data.queries.only")
-      .defaultValue("false")
-      .markAdvanced()
-      .sinceVersion("0.14.0")
-      .withDocumentation("Improves query performance, but queries cannot use hudi metadata fields");
```

Comment:
I added this config when I had the task of speeding up bootstrap reading. What it did was not do the skeleton base file stitching and don't return the meta fields. With this pr, it isn't necessary anymore, because now if you don't request any meta cols, that will automatically happen inside the new reader so I think it is just confusing to introduce this config and then have it not be necessary very soon.

Reply:
Got it.  Could you update the config documentation?  As discussed, we'll keep this config since it can still be used when the existing file format and relations are used.

Reply:
btw, the config doc update should be in a separate PR.